doc/boot_protocol.rst

```diff
@@ -0,0 +1,274 @@
+Security Kernel Loader
+######################
+
+The Security Kernel Loader (SKL) is an implementation of a Secure Loader for
+the SKINIT CPU instruction. The SKL loader takes a minimal approach relying on
+the DRTM Preamble, e.g. the system bootloader, to load into memory and
+configure the Security Kernel that the SKL will measure and execute.
+
+Memory Layout
+*************
+
+The DRTM Preamble is responsible for allocating a contiguous block of physical
+memory and will ensure it is marked as reserved in the memory map. The
+necessary size for the allocation will be found in the SKL header. The DRTM
+Preamble will place the SKL at the beginning of the allocation.
+
+Below is the layout of how the memory allocation is segmented::
+
+      +---     Security Kernel Loader    ---+
+      |                                     |
+      v                                     v
+      +--------+----------+-------------+-------------+--------------+
+      | Header |   Code   |  Boot Tags  | IOMMU Table |  Log Buffer  |
+      +--------+----------+-------------+-------------+--------------+
+      ^                                 ^                            ^
+      |                                 |                            |
+      +                                 +                            +
+  sklbuf_base                          60kb                   header.skl_size
+
+
+SKL Header
+==========
+
+The SKL header is the AMD Secure Loader Block (SLB) header extended to meet the
+needs of the SKL. The first two fields are mandated by the SLB and the SKL adds
+two additional fields which can be seen in the following representative struct,
+::
+
+    typedef struct __packed sl_header {
+        u16 entry_point;        /* Code Entry */
+        u16 boot_tags_offset;   /* RoT Measurement Length */
```

Comment:
These should be the SLB names, and in particular the second one is the measurement length, not the boot tags offset.

We do not want to tie ourselves to an ABI where the boot tags immediately follow the end of measurement.  It happens to be how we lay things out now, but we really ought to have a 3rd pointer to the tags so we can alter the layout if needs be.

Reply:
I was split on this one. Since it was currently set to boot params, I left it for the time being to get comments. I do like the suggestion of adding a field to point to the boot tags.

Reply:
I like the idea for moving the stack, which also goes back to @andyhhp point that for external entities, i.e. bootloader, they do not need to know those details, they need to know how much memory to reserve, and where the boot tags start so they can fill in the appropriate information.